optuna/study/study.py

```diff
@@ -45,6 +47,26 @@
 _logger = logging.get_logger(__name__)
 
 
+def _convert_positional_args(n_positional_args: int = 0) -> Any:
+    def decorator(func: Any) -> Any:
+        @wraps(func)
+        def wrapper(*args: Any, **kwargs: Any) -> Any:
```

Comment:
In my understandingm `func` assumes method or function. How about specifying the type as `Callable`?
```suggestion
def _convert_positional_args(n_positional_args: int = 0) -> Callable:
    def decorator(func: Callable) -> Callable:
        @wraps(func)
        def wrapper(*args: Any, **kwargs: Any) -> Any:
```